homeassistant/components/tesla_fleet/climate.py

```diff
@@ -0,0 +1,300 @@
+            ),
+            (
+                TeslaFleetCabinOverheatProtectionEntity(
+                    vehicle, entry.runtime_data.scopes
+                )
+                for vehicle in entry.runtime_data.vehicles
+            ),
+        )
+    )
+
+
+class TeslaFleetClimateEntity(TeslaFleetVehicleEntity, ClimateEntity):
+    """Tesla Fleet vehicle climate entity."""
+
+    _attr_precision = PRECISION_HALVES
+
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_hvac_modes = [HVACMode.HEAT_COOL, HVACMode.OFF]
+    _attr_supported_features = (
+        ClimateEntityFeature.TURN_ON
+        | ClimateEntityFeature.TURN_OFF
+        | ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.PRESET_MODE
+    )
+    _attr_preset_modes = ["off", "keep", "dog", "camp"]
+    _enable_turn_on_off_backwards_compatibility = False
+
+    def __init__(
+        self,
+        data: TeslaFleetVehicleData,
+        side: TeslaFleetClimateSide,
+        scopes: Scope,
+    ) -> None:
+        """Initialize the climate."""
+        self.scoped = Scope.VEHICLE_CMDS in scopes
+        if not self.scoped or data.signing:
+            self._attr_supported_features = ClimateEntityFeature(0)
+
+        super().__init__(
+            data,
+            side,
+        )
+
+    def _async_update_attrs(self) -> None:
+        """Update the attributes of the entity."""
+        value = self.get("climate_state_is_climate_on")
+        if value is None:
+            self._attr_hvac_mode = None
+        elif value:
+            self._attr_hvac_mode = HVACMode.HEAT_COOL
+        else:
+            self._attr_hvac_mode = HVACMode.OFF
+
+        self._attr_current_temperature = self.get("climate_state_inside_temp")
+        self._attr_target_temperature = self.get(f"climate_state_{self.key}_setting")
+        self._attr_preset_mode = self.get("climate_state_climate_keeper_mode")
+        self._attr_min_temp = cast(
+            float, self.get("climate_state_min_avail_temp", DEFAULT_MIN_TEMP)
+        )
+        self._attr_max_temp = cast(
+            float, self.get("climate_state_max_avail_temp", DEFAULT_MAX_TEMP)
+        )
+
+    async def async_turn_on(self) -> None:
+        """Set the climate state to on."""
+
+        self.raise_for_signing()
+        self.raise_for_scope(Scope.VEHICLE_CMDS)
+        await self.wake_up_if_asleep()
+        await handle_vehicle_command(self.api.auto_conditioning_start())
+
+        self._attr_hvac_mode = HVACMode.HEAT_COOL
+        self.async_write_ha_state()
+
+    async def async_turn_off(self) -> None:
+        """Set the climate state to off."""
+
+        self.raise_for_signing()
+        self.raise_for_scope(Scope.VEHICLE_CMDS)
+        await self.wake_up_if_asleep()
+        await handle_vehicle_command(self.api.auto_conditioning_stop())
+
+        self._attr_hvac_mode = HVACMode.OFF
+        self._attr_preset_mode = self._attr_preset_modes[0]
```

Comment:
Is `climate_state_climate_keeper_mode` preserved/remembered when the vehicle is turned on/off?

Generally, I'm trying to think through what can go wrong given the climate entity is using assumed state to proactive sets attributes that don't come directly from cloud responses (e.g. data update coordinator is no longer the source of truth). I see here this is cleared when turned off but not re-instated when turned on so just want to make sure it works that way where the API clears it on its end also.

Reply:
Preset modes are only used to keep HVAC on after you leave the vehicle, they are never remembered and need to be instantly cleared when HVAC turns off.

By forcing both the HVAC mode and preset mode off at the same time we replicate how the vehicle behavies without having to wait for a coordinator refresh, otherwise Home Assistant would show an impossible state (preset on but HVAC off).

Reply:
Sounds good.